packages/devtools_app/lib/src/provider/instance_viewer/instance_providers.dart

```diff
@@ -0,0 +1,492 @@
+      // objects to have multiple properties with the same name
+      return eval.getInstance(ref, isAlive);
+    },
+    orElse: () => throw FallThroughError(),
+  );
+}
+
+/// Update a variable using the `=` operator.
+///
+/// In rare cases, it is possible for this function to mutate the wrong property.
+/// This can happen when an object contains multiple fields with the same name
+/// (such as private properties or overriden properties), where the conflicting
+/// fields are both defined in the same library.
+Future<void> _mutate(
+  String newValueExpression, {
+  @required InstancePath path,
+  @required AutoDisposeProviderReference ref,
+  @required IsAlive isAlive,
+  @required InstanceDetails parent,
+}) async {
+  await parent.maybeMap(
+    list: (parent) {
+      final eval = ref.watch(evalProvider);
+      final indexPath = path.pathToProperty.last as ListIndexPath;
+      return eval.safeEval(
+        'parent[${indexPath.index}] = $newValueExpression',
+        isAlive: isAlive,
+        scope: {
+          'parent': parent.instanceRefId,
+        },
+      );
+    },
+    map: (parent) {
+      final eval = ref.watch(evalProvider);
+      final keyPath = path.pathToProperty.last as MapKeyPath;
+      final keyRefVar = keyPath.ref == null ? 'null' : 'key';
+
+      return eval.safeEval(
+        'parent[$keyRefVar] = $newValueExpression',
+        isAlive: isAlive,
+        scope: {
+          'parent': parent.instanceRefId,
+          if (keyPath.ref != null) 'key': keyPath.ref,
+        },
+      );
+    },
+    // TODO test can mutate properties of a mixin placed in a different library that the class that uses it
+    object: (parent) {
+      final propertyPath = path.pathToProperty.last as PropertyPath;
+
+      final field =
+          parent.fields.firstWhere((f) => f.name == propertyPath.name);
+
+      return field.eval.safeEval(
+        'parent.${propertyPath.name} = $newValueExpression',
+        isAlive: isAlive,
+        scope: {
+          'parent': parent.instanceRefId,
+        },
+      );
+    },
+    orElse: () => throw StateError('Can only mutate lists/maps/objects'),
+  );
+
+  // TODO(rrousselGit): call notifyListeners/setState/notifyClients based on the modified object
+
+  // Since the same object can be used in multiple locations at once, we need
+  // to refresh the entire tree instead of just the node that was modified.
+  unawaited(ref.container.refresh(rawInstanceProvider(path.root)));
+
+  // Forces the UI to rebuild after the state change
+  await serviceManager.performHotReload();
+}
+
+Future<InstanceDetails> _resolveParent(
+  AutoDisposeProviderReference ref,
+  InstancePath path,
+) async {
+  return path.pathToProperty.isNotEmpty
+      ? await ref.watch(rawInstanceProvider(path.parent).future)
+      : null;
+}
+
+/// Public properties first, then sort alphabetically
+int _sortFieldsByName(ObjectField a, ObjectField b) {
```

Comment:
Move this to utils and apply it to the debugger as well to keep field order consistent.

Reply:
Done